### ISSUE TITLE
🐛 Failing LSP requests after Format on save

### ISSUE THREAD

**Issue by MichaReiser** [rome/tools#3187]
### Environment information

```bash
* v0.8.0
```

### What happened?

just noticed a bug? in the lsp server. when i do typescript renames or vscode replace-on-files, it will attempt to call the formatter as i have format on save. when that happens a bunch of requests are made that fail as it's not stored in the workspace buffer.

https://discord.com/channels/678763474494423051/678763474930761739/1016302822695895140

### Expected result

Requests succeed

### Code of Conduct

- [X] I agree to follow Rome's Code of Conduct

**Comment by MichaReiser** [comment]
@sebmck would you mind providing some additional information about the issue? It could help us reproducing the issue and assessing its priority

Reply from @leops 

> It would need to look into it a bit more, we had some synchronization issues in the past since our implementation of the language server is using full document sync instead of incremental document sync. Because the editor would have to send the whole text buffer on every change, these notifications are debounced or deferred until an actual request is made to the server by the client (like formatting), but the client library used to have a bug where the formatting request ended up being sent before the change notification
It could also be caused by the server processing requests asynchronously, in the case of a workspace-wide find and replace operation the client will send a batch of open notifications for all the changed files, then a batch of formatting requests, then a batch of close notifications, but since there's many of these they might be processed in parallel and possibly out of order on the server (one thread could pick up the formatting task for a file before the corresponding open task has finished processing on another thread) [source](https://discord.com/channels/678763474494423051/678763474930761739/1016636751000895528)
